Tests/MapboxMapsTests/Foundation/MapboxMapTests.swift

```diff
@@ -0,0 +1,76 @@
+import XCTest
+@testable import MapboxMaps
+
+final class MapboxMapTests: XCTestCase {
+
+    var mapClient: MockMapClient!
+    var mapInitOptions: MapInitOptions!
+    var mapboxMap: MapboxMap!
+
+    override func setUp() {
+        super.setUp()
+        let size = CGSize(width: 100, height: 200)
+        mapClient = MockMapClient()
+        mapClient.getMetalViewStub.defaultReturnValue = MTKView(frame: CGRect(origin: .zero, size: size))
+        mapInitOptions = MapInitOptions(mapOptions: MapOptions(size: size))
+        mapboxMap = MapboxMap(mapClient: mapClient, mapInitOptions: mapInitOptions)
+    }
+
+    override func tearDown() {
+        mapboxMap = nil
+        mapInitOptions = nil
+        mapClient = nil
+        super.tearDown()
+    }
+
+    func testInitializationOfResourceOptions() {
+        let actualResourceOptions = mapboxMap.__map.getResourceOptions()
+
+        XCTAssertEqual(actualResourceOptions, mapInitOptions.resourceOptions)
+    }
+
+    func testInitializationOfMapOptions() {
+        let expectedMapOptions = MapOptions(
+            __contextMode: nil,
+            constrainMode: NSNumber(value: mapInitOptions.mapOptions.constrainMode.rawValue),
+            viewportMode: mapInitOptions.mapOptions.viewportMode.map { NSNumber(value: $0.rawValue) },
+            orientation: NSNumber(value: mapInitOptions.mapOptions.orientation.rawValue),
+            crossSourceCollisions: NSNumber(value: mapInitOptions.mapOptions.crossSourceCollisions),
+            size: mapInitOptions.mapOptions.size.map(Size.init),
+            pixelRatio: mapInitOptions.mapOptions.pixelRatio,
+            glyphsRasterizationOptions: nil) // __map.getOptions() always returns nil for glyphsRasterizationOptions
```

Comment:
> `__map.getOptions() always returns nil for glyphsRasterizationOptions`

Is this expected behavior @alexshalamov?

I was hoping that the input map options when creating the Map would be equal to the output when getting the options from the Map. It would certainly clean up this test :)